ntbs-service/Pages/Migration/Index.cshtml.cs

```diff
@@ -33,10 +48,49 @@ public IActionResult OnGetAsync()
 
         public async Task<IActionResult> OnPostAsync()
         {
+            if (!ModelState.IsValid)
+            {
+                return Page();
+            }
+
             var requestId = HttpContext.TraceIdentifier;
 
-            Results = await _service.ImportNotificationsAsync(requestId, NotificationId);
+            if (UploadedFile != null)
+            {
+                var notificationIds = await GetIdListFromFile(UploadedFile);
+                var IdGroups = splitList(notificationIds, 10);
+
+                foreach (var idGroup in IdGroups)
+                {
+                    BackgroundJob.Enqueue<INotificationImportService>(x => x.ImportNotificationsByIdAsync(requestId, idGroup));
+                }
+            }
+            else if (CutoffNotificationDate != null)
+            {
+                CutoffNotificationDate.TryConvertToDateTimeRange(out DateTime? dateRangeStart, out DateTime? dateRangeEnd);
+                Results = await _service.ImportNotificationsByDateAsync(requestId, (DateTime) dateRangeStart);
+            }
+
             return Page();
         }
+
+        private async Task<List<string>> GetIdListFromFile(IFormFile file)
+        {
+            var ids = new List<string>();
+            using (var reader = new StreamReader(file.OpenReadStream()))
+            {
+                while (reader.Peek() >= 0)
+                    ids.Add(await reader.ReadLineAsync()); 
+            }
+            return ids;
+        }
+
+        public static IEnumerable<List<T>> splitList<T>(List<T> ids, int nSize=1000)  
+        {        
+            for (int index = 0; index < ids.Count; index += nSize) 
+            { 
+                yield return ids.GetRange(index, Math.Min(nSize, ids.Count - index)); 
+            }  
+        } 

```

Comment:
Gah, how does LINQ not provide a method like this... Here I was thinking it's a  nice complete package :P

Reply:
`.Skip().Take()` could be used for these purposes - but I don't prefer it over this implementation 
That is assuming we're ever going to enumerate over splitList - at present we're only getting the first 1000 - so `.Take(1000)` would be sufficient (have a comment above to this end).

Reply:
No, I agree this is the right way, I just wanted to vent about linq a little :P

Reply:
I don't think we are using just the first 1000 - we're running a job for each batch of 1000 notifications (see the foreach loop)

Reply:
I was unfortunately afflicted by yield blindness here... :/